packages/manager-components/src/components/public-cloud-project-provider/PublicCloudProjectProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext } from 'react';
+
+import { getIsDiscoveryProject } from './util';
+import { PublicCloudProject } from './publicCloudProject';
+
+const PublicCloudProjectContext = createContext(null);
+
+export const PublicCloudProjectProvider = ({
+  project,
+  children,
+}: {
+  project: PublicCloudProject;
+  children: JSX.Element;
+}) => {
+  return (
+    <PublicCloudProjectContext.Provider value={project}>
+      {children}
+    </PublicCloudProjectContext.Provider>
+  );
+};
+
+export const usePublicCloudProject = () => {
+  return useContext(PublicCloudProjectContext);
+};
```

Comment:
Why create a hook to just call another hook ?

Reply:
The idea behind creating this provider is to,
1. provide an opportunity to developer to set the project information.
2. get the project information or any related information like `isDiscoveryProject` from this provider.

So, this implementation is done to mask any react or other external library specific implementations and also provide an easy way of integration.
Without this utility, developer will have to import `useContext` from react + `PublicCloudProjectContext` from manager-components.